src/lol-ranks.js

```diff
@@ -159,21 +164,34 @@ class LoLRanks {
       if (summonerID !== player.summonerID) {
         reply += i18n.__('reply1') + '\n\n';
 
-        authenticated = false;
-        this.updatePlayer(discordID, { authenticated: false, summonerID });
+        auth = false;
+        this.updatePlayer(discordID, { auth: false, summonerID });
         player = this.getPlayer(discordID);
 
-        await this.removeAllEloRolesFromUser(discordID);
+        await this.removeAllEloRolesFromUser(member);
       }
 
-      if (!authenticated && this.config.enableVerification) {
+      if (player.tier !== null) {
+        const elos = Object.values(i18n.__('ranks'));
+        for (const elo of elos) {
+          if (elo !== player.tier) {
+            const role = member.roles.cache.some(r => r.name === elo);
+            if (role) {
+              const removeRole = member.roles.cache.find(r => r.name === elo);
+              await member.roles.remove(removeRole.id);
+            }
+          }
+        }
+      }
+
+      if (!auth && this.config.enableVerification) {
         try {
           const authData = await this.checkAuth(summonerID);
 
           if (authData === player.authCode) {
             reply += i18n.__('reply2') + '\n\n';
-            authenticated = true;
-            this.updatePlayer(discordID, { authenticated: true });
+            auth = true;
+            this.updatePlayer(discordID, { auth: true });
```

Comment:
I am not a big fan of  changing json config values for nothing..... 
But yeah you don't need to change it.